design/checkpoint_03/harmony_py.ipynb

```diff
@@ -0,0 +1,286 @@
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Notes from a previous meeting:\n",
+    "# extend CMR library to UMM-S; nice output\n",
+    "# when and where do we use Harmony's Capabilities documents?\n",
+    "# stick with Python data structures (dicts, lists, etc.)\n",
+    "# output of CMR should be acceptable input to Harmony python lib\n",
+    "# but also allow a user to submit strings as input for Harmony python lib\n",
+    "# understand UMM-Var response (coupled ot their metadata format)\n",
+    "\n",
+    "# More Notes: We may want to contribute to the CMR python library in order to make feeding data into the Harmony python library easier than what's shown here.\n",
+    "\n",
+    "import re\n",
+    "\n",
+    "# Import CMR's python library\n",
+    "import cmr.search.collection as coll\n",
+    "import cmr.search.granule as gran\n",
+    "\n",
+    "cmr_res = coll.search({'keyword':'MOD09*',\n",
+    "                       'archive_center': 'lp daac'})\n",
+    "\n",
+    "# regex uses a negative look-around assertion\n",
+    "brief = [[r['meta']['concept-id'],\n",
+    "          r['umm']['ShortName'],\n",
+    "          r['meta']['native-id']] for r in cmr_res if re.search('^((?!mmt_collection_).)*$', r['meta']['native-id'])]\n",
+    "[print(b) for b in brief]\n",
+    "# ['C193529903-LPDAAC_ECS', 'MOD09GQ', 'MODIS/Terra Surface Reflectance Daily L2G Global 250m SIN Grid V006']\n",
+    "# ['C193529902-LPDAAC_ECS', 'MOD09GA', 'MODIS/Terra Surface Reflectance Daily L2G Global 1km and 500m SIN Grid V006']\n",
+    "# ['C193529899-LPDAAC_ECS', 'MOD09A1', 'MODIS/Terra Surface Reflectance 8-Day L3 Global 500m SIN Grid V006']\n",
+    "# ['C193529944-LPDAAC_ECS', 'MOD09Q1', 'MODIS/Terra Surface Reflectance 8-Day L3 Global 250m SIN Grid V006']\n",
+    "# ['C193529901-LPDAAC_ECS', 'MOD09CMG', 'MODIS/Terra Surface Reflectance Daily L3 Global 0.05Deg CMG V006']\n",
+    "\n",
+    "\n",
+    "####\n",
+    "# The CMR python library does not support variable browsing at this time.\n",
+    "####\n",
+    "\n",
+    "\n",
+    "# The output from CMR may be used as input to the Harmony python library\n",
+    "req = Request(\n",
+    "    collection=cmr_res[0],\n",
+    "    spatial=Bbox(lat=(40, 42), lon=(-107, -105)),  # could accept bbox, shapely, geojson (polygon)\n",
+    "    temporal=Temporal(start=datetime.date(2020, 6, 1), stop=datetime.date(2020, 6, 30)),\n",
+    "    format=Format.ZARR\n",
+    ")\n"
+   ]
+  },
+  {
+   "source": [
+    "## Async vs. Sync Request Submit()"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sync Request\n",
+    "res = Request.submit(req, auth, sync=True)\n",
```

Comment:
Just to avoid any confusion for external review, you may want to change this and all following examples to fit with the client concept:

`res = client.submit(request, verbose=True, sync=True)`

Reply:
Thank you for catching that - I overlooked those after changing the summary usage.